staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -257,7 +260,7 @@ func (rules *ClientConfigLoadingRules) Migrate() error {
 		return nil
 	}
 
-	for destination, source := range rules.MigrationRules {
+	for source, destination := range rules.MigrationRules {
```

Comment:
this wasn't an issue before, because there was a single item in the list, but iteration order over a map is poorly defined

Reply:
@liggitt thanks for your input. I was afraid of that. Other way would be: just replace the migration "~/.kube/.kubeconfig" as default value to ~/.kube/config and this would bring a minimum change or still valid migration path that we need to care?

It seems old issue now: https://github.com/kubernetes/kubernetes/issues/4615